src/components/user/user.service.ts

```diff
@@ -433,27 +448,42 @@ export class UserService {
       .query()
       .call(matchRequestingUser, session)
       .call(matchUserPermissions, 'User', id)
-      .call(addAllSecureProperties, ...props)
+      .call(addAllSecureProperties, ...[...props, 'roles'])
       .with([
         ...props.map(addPropertyCoalesceWithClause),
+        `
+        {
+          value: collect(distinct roles.value),
+          canRead: coalesce(rolesReadPerm.read, false),
+          canEdit: coalesce(rolesEditPerm.edit, false)
+        } as roles
+        `,
         'coalesce(node.id) as id',
         'coalesce(node.createdAt) as createdAt',
       ])
-      .returnDistinct([...props, 'id', 'createdAt']);
+      .returnDistinct([...props, 'roles', 'id', 'createdAt']);
 
     const result = (await query.first()) as User | undefined;
     if (!result) {
       throw new NotFoundException('Could not find user');
     }
 
-    return result;
+    const response: any = {
+      ...result,
+      roles: {
+        value: result.roles.value || [],
+        canRead: result.roles.canEdit ?? result.roles.canRead,
```

Comment:
This doesn't seem right. Why would canRead come from canEdit first? This prevents a read-only view of this property. Maybe what you meant was `||` instead of `??` which would fallback to canRead if canEdit was false. Currently if canEdit false canRead is never used.

Reply:
This makes canRead true if canEdit is true. If canEdit is false then canRead may or may not be true so it returns directly. The principle is if canEdit is true, we'll assume that canRead should be true.

Reply:
Ok this isn't doing what you think it is then. The second value, canRead, will never be used here - only `canEdit`. If canEdit was nullable that would be a different story, but it's not - it's always a boolean.
```tsx
(false ?? true) // false
```

Reply:
that's right. my mistake. this could have been done in the cypher `WITH` clause where `canEdit` might be null.